tests/testsuite/tool_paths.rs

```diff
@@ -352,3 +352,19 @@ fn cfg_ignored_fields() {
         )
         .run();
 }
+
+#[cargo_test]
+fn env_var_contain_lower_case() {
+    let p = project().file("src/main.rs", "fn main() {}").build();
+
+    let env_key = "CARGO_TARGET_X86_64_UNKNOWN_LINUX_musl_LINKER";
+
+    p.cargo("build -v --target X86_64_unknown_linux_musl")
```

Comment:
This target name should be `x86_64-unknown-linux-musl`, I think. As far as I can tell, this is an invalid target name passed to `cargo`, and will produce an error (in addition to this warning).

Reply:
From what i can tell, ConfigKey::push replaces the dashes with unders and upper-case the target. When it gets to Config::has_key, the key.as_env_key() yield correctly  formatted key for triple.  

```
pub fn push(&mut self, name: &str) {
        let env = name.replace("-", "_").to_uppercase();
        self._push(&env, name);
    }
```

Reply:
I think what Josh is saying here is that `X86_64_unknown_linux_musl` looks almost correct, but it is in fact an invalid target name.  This should, instead, use `rustc_host()`, and in the env var.